ksqldb-common/src/main/java/io/confluent/ksql/util/BytesUtils.java

```diff
@@ -110,6 +112,83 @@ public static String encode(final byte[] value, final String encoding) {
     return Arrays.copyOfRange(getByteArray(buffer), start, end);
   }
 
+  public static List<byte[]> split(final byte[] b, final byte[] delim) {
+    if (b.length == 0) {
+      return Arrays.asList(b);
+    } else if (delim.length == 0) {
+      return splitAllBytes(b);
+    }
+
+    int offset = 0;
+    int delimIdx;
+
+    final List<byte[]> list = new ArrayList<>();
+    while ((delimIdx = indexOf(b, delim, offset)) != -1) {
+      final int newSplitLength = delimIdx - offset;
+      final byte[] newSplitArray = new byte[newSplitLength];
+
+      System.arraycopy(b, offset, newSplitArray, 0, newSplitLength);
+      list.add(newSplitArray);
+
+      offset = delimIdx + delim.length;
+    }
+
+    if (offset == 0) {
+      return Arrays.asList(b);
+    } else {
+      list.add(Arrays.copyOfRange(b, offset, b.length));
```

Comment:
If the offset is 0, then `list` would be empty and `Arrays.copyOfRange(b, offset, b.length)` would just return  `b`, so we can skip the first condition.